sdk/dotnet/Pulumi/Serialization/Serializer.cs

```diff
@@ -134,25 +134,56 @@ public Serializer(bool excessiveDebugOutput)
                 {
                     Log.Debug($"Serialize property[{ctx}]: Recursing into Output");
                 }
-
                 var data = await output.GetDataAsync().ConfigureAwait(false);
                 this.DependentResources.AddRange(data.Resources);
+                var propResources = new HashSet<Resource>();
+                propResources.UnionWith(data.Resources);
```

Comment:
Also, wondering if it'd be more clear named: `propResources` => `outputResources`.

Reply:
It's `propResources` because it makes it easier to look between the C# and TypeScript implementations. That's the variable name that `rpc.ts` uses.